src/dynarec/la64/dynarec_la64_660f.c

```diff
@@ -1225,6 +1240,33 @@ uintptr_t dynarec64_660F(dynarec_la64_t* dyn, uintptr_t addr, uintptr_t ip, int
             GWBACK;
             UFLAG_DF(x1, d_imul16);
             break;
+        case 0xB6:
```

Comment:
Use bits extract and pick instructions to simplify this, refer the the arm64 implementation.